### PR TITLE
docs: move json functions to single file

### PR DESCRIPTION
**Pull request by CookiePieWw**
## What's Changed in this PR

Move json related functions to a single file as geo functions.

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

**Comment by killme2008** [comment]
Good job! Don't forget sidebar https://github.com/GreptimeTeam/docs/blob/main/sidebars.ts

**Comment by CookiePieWw** [comment]
Mostly are copied. Add some title:

1. Converter/转换: for `parse_json` and `json_to_string`
2. Extractor/提取: for `json_get_<type>`
3. Validation/验证: for `json_is_<type>` and `json_path_exists`

**Comment by CookiePieWw** [comment]
> Good job! Don't forget sidebar https://github.com/GreptimeTeam/docs/blob/main/sidebars.ts

Thanks for reminding :)

File: docs/reference/sql/functions/json.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import TOCInline from '@theme/TOCInline';
```
Comment (nicecui) [inline docs/reference/sql/functions/json.md:1]:
```suggestion

```

File: docs/reference/sql/functions/json.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+# JSON Functions
+
+This page lists all json type related functions in GreptimeDB.
+
+<TOCInline toc={toc} />
```
Comment (nicecui) [inline docs/reference/sql/functions/json.md:7]:
```suggestion

```

File: i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/json.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
+import TOCInline from '@theme/TOCInline';
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/json.md:1]:
```suggestion
```

File: i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/json.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+# JSON 函数
+
+本页面列出了 GreptimeDB 中所有 JSON 类型相关的函数。
+
+<TOCInline toc={toc} />
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/reference/sql/functions/json.md:7]:
```suggestion
```

**Review by nicecui — APPROVED** [review]
There is a sub-navigation on the right side of the document page, so there is no need to add navigation manually

**Review by WenyXu — APPROVED** [review]

**Pull request merged**